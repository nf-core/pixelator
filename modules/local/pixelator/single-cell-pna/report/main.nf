process PIXELATOR_PNA_REPORT {
    tag "${meta.id}"
    label 'process_low'


    // TODO: Add conda
    // conda "bioconda::pixelator=0.18.2"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/pixelgen-technologies/pixelator:0.21.2'
        : 'quay.io/pixelgen-technologies/pixelator:0.21.2'}"

    input:
    tuple val(meta), path(panel_file), val(panel)
    path amplicon_data,      stageAs: "results/amplicon/*"
    path demux_data,         stageAs: "results/demux/*"
    path collapse_data,      stageAs: "results/collapse/*"
    path graph_data,         stageAs: "results/graph/*"
    path analysis_data,      stageAs: "results/analysis/*"
    path post_analysis_data, stageAs: "results/post_analysis/*"
    path layout_data,        stageAs: "results/layout/*"

    output:
    tuple val(meta), path("report/*.html"),    emit: report
    tuple val(meta), path("*pixelator-*.log"), emit: log

    path "versions.yml",                       emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''
    def panelOpt = (panel
        ? "--panel ${panel}"
        : panel_file
            ? "--panel ${panel_file}"
            : "")

    """
    pixelator \\
        --cores ${task.cpus} \\
        --log-file ${prefix}.pixelator-report.log \\
        --verbose \\
        single-cell-pna \\
        report \\
        --output . \\
        ${panelOpt} \\
        ${args} \\
        results

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir report
    touch report/${prefix}.report.html
    touch ${prefix}.pixelator-report.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """
}
