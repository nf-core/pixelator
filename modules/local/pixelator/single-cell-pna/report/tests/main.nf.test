nextflow_process {

    name "Test Process PIXELATOR_PNA_REPORT"
    script "../main.nf"
    process "PIXELATOR_PNA_REPORT"
    tag "modules"
    tag "pixelator"
    tag "pixelator/pna"
    tag "pixelator/single_cell_pna_report"

    test("PNA report - small test") {
        when {
            process {
                """
                input[0] = [
                    [ id:'PNA055_Sample07_filtered_S7', design:'pna-2', panel:'proxiome-immuno-155', technology:'pna' ],
                    [],
                    'proxiome-immuno-155',
                ]
                input[1] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/amplicon/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/amplicon/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                ]
                input[2] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/demux/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/demux/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                ]
                input[3] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/collapse/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                ]
                input[4] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/graph/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/graph/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/graph/PNA055_Sample07_filtered_S7.graph.pxl', checkIfExists: true),
                ]
                input[5] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/analysis/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/analysis/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                ]
                input[6] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/post_analysis/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/post_analysis/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                ]
                input[7] = [
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/layout/PNA055_Sample07_filtered_S7.meta.json', checkIfExists: true),
                    file(params.pipelines_testdata_base_path + 'testdata/pna/modules/layout/PNA055_Sample07_filtered_S7.report.json', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                {
                    with(process.out.report) {
                        with(get(0)) {
                            assert get(1).endsWith("qc-report.html")
                            assert file(get(1)).exists()
                        }
                    }
                },
                { assert snapshot(process.out.versions).match("versions") },
            )
        }
    }
}

