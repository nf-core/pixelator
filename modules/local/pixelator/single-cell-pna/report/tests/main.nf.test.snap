{
    "versions": {
        "content": [
            [
                "versions.yml:md5,277fdd55773113498ea1c9177f927674"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-26T14:02:28.335957996"
    }
}